### PR TITLE
Rename network authentication and encryption variables for configuration.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This renames two variables in the network configuration to avoid confusion. These variable renaming changes are: (i) 'is_permissioned' -> 'enable_remote_authentication'; and (ii) 'enable_encryption_and_authentication' -> 'enable_noise'. 

(The old variable names were somewhat misleading!)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Local runs of 'cargo xtest' and libra-swarm seem to produce the expected output.

## Related PRs

None.

### CODE REVIEW

File: config/src/config/network_config.rs [hunk 1]
```diff
@@ -35,17 +35,17 @@ pub struct NetworkConfig {
     pub discovery_interval_ms: u64,
     pub connectivity_check_interval_ms: u64,
     // Flag to toggle if Noise is used for encryption and authentication.
-    pub enable_encryption_and_authentication: bool,
-    // If the network is permissioned, only trusted peers are allowed to connect. Otherwise, any
-    // node can connect. If this flag is set to true, the `enable_encryption_and_authentication`
+    pub enable_noise: bool,
+    // If network authentication is enabled, only trusted peers are allowed to connect. Otherwise, any
+    // node can connect. If this flag is set to true, the `enable_noise`
```
Comment (davidiw) [inline config/src/config/network_config.rs]: the should be removed before `enable_noise`

File: config/src/config/network_config.rs [hunk 2]
```diff
@@ -140,10 +140,10 @@ impl NetworkConfig {
                 self.network_peers.peers.is_empty(),
                 "Validators should not define network_peers"
             );
-        } else if self.is_permissioned {
+        } else if self.enable_remote_authentication {
```
Comment (davidiw) [inline config/src/config/network_config.rs]: there's actually no way to enforce ids outside of the other peer enforcing remote authentication, we should consider deleting this piece of code...  we should consider deleting this code. cc: @phlip9

Comment (davidiw) [inline config/src/config/network_config.rs]: please add a check in this function that ensure noise is enabled if enable_remote_authentication is

File: libra-node/src/main_node.rs [hunk 3]
```diff
@@ -120,11 +120,11 @@ pub fn setup_network(
...
         ]);
-    if config.is_permissioned {
-        // If the node wants to run in permissioned mode, it should also have authentication and
-        // encryption.
+    if config.enable_remote_authentication {
+        // If the node wants to run in enable_remote_authentication mode, it should also use noise for
+        // authentication and encryption.
         assert!(
-            config.enable_encryption_and_authentication,
+            config.enable_noise,
```
Comment (davidiw) [inline libra-node/src/main_node.rs]: better to assert in the config code so that we can validate configs in the config code only

Comment (JoshLind) [reply libra-node/src/main_node.rs]: As discussed offline, we'll address these logic changes in a future PR. This PR is just for renaming.

File: config/src/config/network_config.rs (hunk 2 above)
Comment (JoshLind) [reply config/src/config/network_config.rs]: As discussed offline, we'll address these logic changes in a future PR. This PR is just for renaming.

Comment (JoshLind) [reply config/src/config/network_config.rs]: As discussed offline, we'll address these logic changes in a future PR. This PR is just for renaming.

**Comment by JoshLind** [comment]
Thanks @davidiw! It's ready for another look 😄

File: config/src/config/network_config.rs (hunk 1 above)
Comment (JoshLind) [reply config/src/config/network_config.rs]: Done, thanks!

**Comment by JoshLind** [comment]
Thanks @davidiw! It's ready for another look 😄

**Comment by davidiw** [comment]
Looks like you have a conflict you need to fix before landing.

**Comment by JoshLind** [comment]
@davidiw, thanks for pointing that out! I've rebased from master and fixed the conflicts. If you're happy with the changes, any chance you could run the libra-bors command? Hopefully we get it fixed next week so that I have appropriate permissions!

**Pull request closed without merging**